[JBPM-10218] Add italian translation

gjed opened the pull request:
Add support for italian translation

**JIRA**: [JBPM-10218](https://issues.redhat.com/browse/JBPM-10218)

**Referenced Pull Requests**:
* [appformer#1413](https://github.com/kiegroup/appformer/pull/1413)
* [drools-wb#1573](https://github.com/kiegroup/drools-wb/pull/1573)
* [jbpm-wb#1604](https://github.com/kiegroup/jbpm-wb/pull/1604)
* [kie-wb-distributions#1239](https://github.com/kiegroup/kie-wb-distributions/pull/1239)

There is still a bit missing: the italian language does not appear in the dropdown menu in the setting page. I edited the LanguageConfigurationHandler, WorkbenchConfigurationPopup and CommonCostants classes and the gwt.xml files but something is still missing. Any advice is kindly accepted!

---
<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

gjed commented:
Rebased gjed/i18n_it on main branch

mareknovotny commented:
you also need to add Italian into constants and java classes which populate dropdowns ;)
see for inspiration this PR change from past https://github.com/kiegroup/kie-wb-common/pull/3191/files

gjed commented:
> you also need to add Italian into constants and java classes which populate dropdowns ;) see for inspiration this PR change from past https://github.com/kiegroup/kie-wb-common/pull/3191/files

Thanks for the answer!
I already added the option in the map of the WorkbenchConfigurationPopup and LanguageConfigurationHandler classes, added the method to the CommonConstants interface and the values in the CommonCostants*.properties are populated. (first commit 2a2a471)
That was in fact inspired on the #3191 PR.

gjed commented:
fyi: the italian option in the language menu is present.

About the build: I'm not sure why it breaks on the `org.kie.server:kie-server-services-drools:jar:7.75.0-SNAPSHOT` dependency.

mareknovotny commented:
jenkins retest this please

pibizza reviewed (changes requested):
I have added a number of inline comments and reviewed almost all the files. There are a number of issues about having consistent translations - like translating or not containers and forms - but these would require a deeper review.

pibizza commented:
jenkins retest this please

mareknovotny commented:
jenkins retest this please

mareknovotny commented on kie-wb-common-services/kie-wb-common-verifier/kie-wb-common-verifier-reporting-client/src/main/java/org/kie/workbench/common/services/verifier/reporting/client/resources/i18n/AnalysisConstants_it.properties:
```diff
@@ -0,0 +1,65 @@
+RedundantRowsP1=La ridondanza esiste quando due righe fanno le stesse azioni con lo stesso insieme di fatti forniti.
+RedundantRowsP2=La ridondanza potrebbe non essere un problema se le regole ridondanti impostano un valore su un fatto esistente, questo imposta il valore due volte. I problemi si verificano quando le due regole aumentano un contatore o aggiungono più fatti nella memoria di lavoro. In entrambi i casi, l''altra riga non è necessaria.
+RedundantRowsP3=Per risolvere il problema, rimuovere una delle righe ridondanti.
+SubsumptantRowsP1=L'inclusione esiste quando una riga fa la stessa cosa di un''altra, con un sottoinsieme dei valori/fatti di un''altra regola.
```
```suggestion
SubsumptantRowsP1=L''inclusione esiste quando una riga fa la stessa cosa di un''altra, con un sottoinsieme dei valori/fatti di un''altra regola.
```

gjed reviewed (commented):
lgtm

mareknovotny commented:
jenkins retest this please
